### PR TITLE
[RPS-183] Modify search to match part words

### PR DESCRIPTION
**Pull request by ngta1**
Include wildcard to search terms to allow matching on partial words. Added unit test to cover various search scenarios.

### CODE REVIEW

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java [hunk 1]
```diff
@@ -84,15 +135,16 @@ protected EssentialsListComponentInfo getComponentInfo(HstRequest request) {
...
         HstQueryBuilder queryBuilder = HstQueryBuilder.create(getSearchScope(request));
 
         // register content classes
         addPublicationSystemTypes(queryBuilder);
 
         return queryBuilder
             .where(or(
-                publicationSystemConstraint(query),
-                constraint(".").contains(query)
+                publicationSystemConstraint(queryIncWildcards),
```
Comment (wojtek-oledzki) [inline site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: I think this should be:

```
.where(or (
  publicationSystemConstraint(query),
  publicationSystemConstraint(queryIncWildcards),
  constraint(".").contains(query),
  constraint(".").contains(queryIncWildcards),
```

**Comment by ngta1** [comment]
thanks, will fix.

**Comment by wojtek-oledzki** [comment]
Looks good to me :tm:

LGTM

**Comment by wojtek-oledzki** [comment]
building locally to test some search queries myself ...

**Pull request merged**